ssl/statem/statem_lib.c

```diff
@@ -1609,14 +1609,16 @@ int ssl_version_supported(const SSL *s, int version, const SSL_METHOD **meth)
     for (vent = table;
          vent->version != 0 && version_cmp(s, version, vent->version) <= 0;
          ++vent) {
-        if (vent->cmeth != NULL
+        const SSL_METHOD *(*ret_meth) (void);
```

Comment:
It may be simpler to just make `ret_meth` a `SSL_METHOD *` directly, and only call the function once, on the next line.

Reply:
The table entry contains function pointers:
```
typedef struct {
    int version;
    const SSL_METHOD *(*cmeth) (void);
    const SSL_METHOD *(*smeth) (void);
} version_info;
```

So we need first to check that the function pointer is not null before calling the function that returns a method. Just like it was before the fix.